backend/src/main/java/io/github/projektalmanac/amoxtli/backend/service/UserService.java

```diff
@@ -305,5 +311,89 @@ public SessionTokenDto iniciarSesion(CredencialesDto credenciales){ //hash passw
+
+        if(!user.getExchangesAccepting().isEmpty() && user.getExchangesAccepting().size() < 4){
+            throw new InvalidUserException("No puedes realizar más de 4 intercambios a la vez.");
+        }
+        ValidaPuedeIntercambiar200ResponseDto response = new ValidaPuedeIntercambiar200ResponseDto();
+        response.setPuedeIntercambiar(true);
+        response.mensaje("Puede solicitar intercambiar un libro.");
+
+        return response;
+    }
+
+    public IntercambioDto solicitaIntercambio(Integer idOfertante, CreacionIntercambioDto creacionIntercambioDto) {
+
+        if(creacionIntercambioDto == null){
+            throw new IllegalArgumentException("El cuerpo de la petición no puede ser nulo.");
+        }
+        // Recuperamos la información del usuario y del libro que recibe el intercambio
+        Integer idAceptante = creacionIntercambioDto.getIdAceptante();
+        Integer idLibroAceptante = creacionIntercambioDto.getIdLibroAceptante();
+
+        if (Objects.equals(idOfertante, idAceptante)){
+            throw new BadRequestException("No puedes intercambiar libros contigo mismo.");
+        }
+        // Obtenemos a los usuarios y al libro que se intercambiará
+        Optional <User> userOptionalOfertante = this.userRepository.findById(idOfertante);
+        if (userOptionalOfertante.isEmpty()) {
+            throw new UserNotFoundException(idOfertante);
+        }
+
+        Optional<User> userOptionalAceptante = this.userRepository.findById(idAceptante);
+        if (userOptionalAceptante.isEmpty()) {
+            throw new UserNotFoundException(idAceptante);
+        }
+
+        // Extraemos el libro que se intercambiará de parte de
+        List<Book> libros = userOptionalAceptante.get().getBooks();
+        Book book = libros.stream()
+                .filter(libro -> libro.getId() == idLibroAceptante)
+                .findFirst()
+                .orElseThrow(() -> new BookNotFoundException("El libro no fue encontrado."));
+
+        Exchange exchange = new Exchange();
+
+        exchange.setStatus(Status.PENDIENTE);
+        exchange.setUserOfferor(userOptionalOfertante.get());
+        exchange.setUserAccepting(userOptionalAceptante.get());
+        exchange.setBookAccepting(book);
+
+        this.exchangeRepository.save(exchange);
+
+        List<Exchange> exchangesOfertante = userOptionalOfertante.get().getExchangesOfferor();
+        exchangesOfertante.add(exchange);
+
+        userOptionalOfertante.get().setExchangesOfferor(exchangesOfertante);
+
+        // Se hace el mapeo de las entidades
+        OfertanteDto ofertanteDto = ExchangeMapper.INSTANCE.toOfertanteDto(userOptionalOfertante.get());
+        AceptanteDto aceptanteDto = ExchangeMapper.INSTANCE.toAceptanteDto(userOptionalAceptante.get());
+        LibroAceptanteDto libroAceptanteDto = ExchangeMapper.INSTANCE.tOLibroAceptanteDto(book);
+
+
+        IntercambioDto intercambioDto = new IntercambioDto();
+        intercambioDto.setId(0);
```

Comment:
```suggestion
        intercambioDto.setId(exchange.getId());
```

Por qué pones el id en 0?